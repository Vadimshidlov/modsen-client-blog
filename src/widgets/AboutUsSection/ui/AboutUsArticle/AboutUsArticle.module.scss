.aboutUsTextContent {
  display: flex;
  flex-direction: column;

  .aboutUsTeamTitle {
    margin-bottom: 16px;
    @include headingTwo;

    @media (max-width: $tablet) {
      @include headingFour;
    }

    @media (max-width: $mobile) {
      @include headingSix;
    }
  }

  .aboutUsTeamSubtitle {
    margin-bottom: 16px;
    max-width: 520px;
    @include headingFour;

    @media (max-width: $tablet) {
      @include headingSix;
    }

    @media (max-width: $mobile) {
      @include headingSeven;
      font-weight: $bold;
    }
  }

  .aboutUsText {
    max-width: 520px;
    @include bodyOne;
  }
}
