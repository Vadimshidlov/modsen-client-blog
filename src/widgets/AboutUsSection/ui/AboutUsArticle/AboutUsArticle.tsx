import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle.module.scss';

export type AboutUsArticlePropsType = {
  title: string;
  subTitle: string;
  text: string;
};

export function AboutUsArticle({ title, subTitle, text }: AboutUsArticlePropsType) {
  return (
    <article className={styles.aboutUsTextContent}>
      <h2 className={styles.aboutUsTeamTitle}>{title}</h2>
      <h4 className={styles.aboutUsTeamSubtitle}>{subTitle}</h4>
      <span className={styles.aboutUsText}>{text}</span>
    </article>
  );
}
