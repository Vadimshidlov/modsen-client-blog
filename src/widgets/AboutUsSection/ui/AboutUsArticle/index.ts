export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
