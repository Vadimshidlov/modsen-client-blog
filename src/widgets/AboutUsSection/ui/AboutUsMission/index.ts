export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
