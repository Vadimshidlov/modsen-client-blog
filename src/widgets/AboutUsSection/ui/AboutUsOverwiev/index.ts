export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
