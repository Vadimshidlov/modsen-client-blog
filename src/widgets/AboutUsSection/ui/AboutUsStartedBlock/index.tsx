export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
