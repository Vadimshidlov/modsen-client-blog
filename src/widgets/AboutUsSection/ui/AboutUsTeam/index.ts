export { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam/AboutUsTeam';
