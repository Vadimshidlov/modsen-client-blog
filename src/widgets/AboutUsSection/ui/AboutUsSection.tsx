import React from 'react';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
