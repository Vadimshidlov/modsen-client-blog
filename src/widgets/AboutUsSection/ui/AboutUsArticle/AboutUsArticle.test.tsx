import React from 'react';
import { render } from '@testing-library/react';
import { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle';

describe('AboutUsArticle component tests', () => {
  const title = 'Title';
  const subTitle = 'Subtitle';
  const text = 'Text content';

  test('should renders with correct content', () => {
    const { getByText } = render(<AboutUsArticle title={title} subTitle={subTitle} text={text} />);

    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(subTitle)).toBeInTheDocument();
    expect(getByText(text)).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { asFragment } = render(<AboutUsArticle title={title} subTitle={subTitle} text={text} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
