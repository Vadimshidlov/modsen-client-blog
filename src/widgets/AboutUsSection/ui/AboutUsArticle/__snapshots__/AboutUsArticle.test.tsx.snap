// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsArticle component tests should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="aboutUsTextContent"
  >
    <h2
      class="aboutUsTeamTitle"
    >
      Title
    </h2>
    <h4
      class="aboutUsTeamSubtitle"
    >
      Subtitle
    </h4>
    <span
      class="aboutUsText"
    >
      Text content
    </span>
  </article>
</DocumentFragment>
`;
